torchmetrics/text/squad.py

```diff
@@ -0,0 +1,198 @@
+        """
+        if isinstance(preds, Dict):
+            preds = [preds]
+
+        if isinstance(targets, Dict):
+            targets = [targets]
+
+        for pred in preds:
+            keys = pred.keys()
+            if "prediction_text" not in keys or "id" not in keys:
+                raise KeyError(
+                    "Expected keys in a single prediction are 'prediction_text' and 'id'."
+                    "Please make sure that 'prediction_text' maps to the answer string and 'id' maps to the key string."
+                )
+
+        for target in targets:
+            keys = target.keys()
+            if "answers" not in keys or "id" not in keys:
+                raise KeyError(
+                    "Expected keys in a single target are 'answers' and 'id'."
+                    "Please make sure that 'answers' maps to a `SQuAD` format dictionary and 'id' maps to the key"
+                    "string.\n"
+                    "SQuAD Format: "
+                    "{"
+                    "    'answers': {"
+                    "        'answer_start': [1],"
+                    "        'text': ['This is a test text']"
+                    "    },"
+                    "    'context': 'This is a test context.',"
+                    "    'id': '1',"
+                    "    'question': 'Is this a test?',"
+                    "    'title': 'train test'"
+                    "}"
+                )
+
+            answers_keys = target["answers"].keys()
+            if "text" not in answers_keys:
+                raise KeyError(
+                    "Expected keys in a 'answers' are 'text'."
+                    "Please make sure that 'answer' maps to a `SQuAD` format dictionary.\n"
+                    "SQuAD Format: "
+                    "{"
+                    "    'answers': {"
+                    "        'answer_start': [1],"
+                    "        'text': ['This is a test text']"
+                    "    },"
+                    "    'context': 'This is a test context.',"
+                    "    'id': '1',"
+                    "    'question': 'Is this a test?',"
+                    "    'title': 'train test'"
+                    "}"
+                )
+
+        preds_dict = {prediction["id"]: prediction["prediction_text"] for prediction in preds}
+        targets_dict = [
+            {
+                "paragraphs": [
+                    {
+                        "qas": [
+                            {
+                                "answers": [{"text": answer_text} for answer_text in target["answers"]["text"]],
+                                "id": target["id"],
+                            }
+                            for target in targets
+                        ]
+                    }
+                ]
+            }
+        ]
+        scores: Tuple[Tensor, Tensor, Tensor] = _squad_update(preds_dict, targets_dict)
```

Comment:
I think `mypy` will infer types. Maybe we can directly unroll the values like: 
```suggestion
        f1_score, exact_match, total = _squad_update(preds_dict, targets_dict)
```